configure.cmake

```diff
@@ -874,8 +881,15 @@ ELSE()
   SET(OLD_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
   LIST(APPEND CMAKE_REQUIRED_LIBRARIES "atomic")
   CHECK_CXX_SOURCE_COMPILES("
+  #include <atomic>
+  std::atomic<int> x;
+  std::atomic<short> y;
```

Comment:
Does it make a difference whether we create 1 or 3 atomic variables to perform the check?

Reply:
I am really not certain about this. It checks for different sizes of __atomic_add_fetch.  Riscv spec specifies atomic ops can be 32 bits or 64 bits (RV64 only) and doesn't support byte increment. So yeah, checking short and char should be equivalent. int is 4 bytes for both rv64 and rv32 machines and should have builtin atomics unless the machine lacks atomic extension.
But I think it is more future proof to add more checks given they aren't very expensive checks.